### PR TITLE
Add and test nightly builds

### LINKED ISSUE #2143: Publish nightly image builds and E2E tests using nightlies

### PR DESCRIPTION
**Pull request by rainest**
**What this PR does / why we need it**:
Nightly builds and E2E tests using nightly builds. The nightly build is available on demand, so we can also run it during the day for release testing.

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #2143

**Special notes for your reviewer**:
This requires coordination with other teams to create the new `kong/nightly-ingress-controller` Docker repo and grant our Docker token permission to it. This must be held until that is ready.

### CODE REVIEW

File: test/e2e/all_in_one_test.go [hunk 1]
```diff
@@ -69,7 +69,10 @@ const (
 	adminAPIWait = time.Minute * 2
 )
 
-var imageOverride = os.Getenv("TEST_KONG_CONTROLLER_IMAGE_OVERRIDE")
+var (
+	imageOverride = os.Getenv("TEST_KONG_CONTROLLER_IMAGE_OVERRIDE")
+	imageLoad     = os.Getenv("TEST_KONG_CONTROLLER_IMAGE_LOAD")
```
Comment (shaneutt) [inline test/e2e/all_in_one_test.go:74]: I know the previous var was missing any godoc, but now that we have two similar vars it seems like a good time to add some godoc here to point out upfront what they're for.

**Review by shaneutt — DISMISSED** [review]
:+1:

**Comment by rainest** [comment]
Per @hutchic this repo is now ready, this can merge after review. No great way to test prior to merge, but no serious consequences if it breaks, so just watch the jobs to see if they fail, then fix if they do.

Needs re-review since I goofed one of the repo targets originally.

**Review by shaneutt — APPROVED** [review]
LGTM, as you say we're limited a bit in our testing but if it's not working quite right at first it's all test code so we can patch after.

**Pull request merged**